From c1bd680e12e56f2eea235d22548e0cf85185ab53 Mon Sep 17 00:00:00 2001
From: hongxu <hongxu.jia@windriver.com>
Date: Mon, 5 Jul 2021 17:30:08 +0800
Subject: [PATCH] skopeo: fix native skopeo failed if no libdevmapper.so.1.02
 on host

If host does not install libdevmapper.so.1.02, run native skopeo failed:
...
$ tmp-glibc/sysroots/x86_64/usr/sbin/skopeo -h
|tmp-glibc/sysroots/x86_64/usr/sbin/skopeo.real: error while loading
shared libraries: libdevmapper.so.1.02: cannot open shared object file: No such file or directory
...

Create wrapper to set LD_LIBRARY_PATH which using native
libdevmapper.so.1.02

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
Signed-off-by: Bruce Ashfield <bruce.ashfield@gmail.com>
---
 recipes-containers/skopeo/skopeo_git.bb | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/recipes-containers/skopeo/skopeo_git.bb b/recipes-containers/skopeo/skopeo_git.bb
index 6c95bf3..19ac938 100644
--- a/recipes-containers/skopeo/skopeo_git.bb
+++ b/recipes-containers/skopeo/skopeo_git.bb
@@ -84,6 +84,9 @@ do_install() {
 do_install_append_class-native() {
     create_cmdline_wrapper ${D}/${sbindir}/skopeo \
         --policy ${sysconfdir}/containers/policy.json
+
+    create_wrapper ${D}/${sbindir}/skopeo.real \
+        LD_LIBRARY_PATH=${STAGING_LIBDIR_NATIVE}
 }
 
 do_install_append_class-nativesdk() {
-- 
2.25.1

